src/bioext/elastic_utils.py

```diff
@@ -11,25 +11,38 @@ def __init__(self, *args, **kwargs) -> None:
         self.session.proxies = {"http": self.proxy_endpoint, "https":self.proxy_endpoint}
 
 class ElasticsearchSession:
-    def __init__(self, server=None):
+    def __init__(self, proxy=None, conn_mode:str="HTTP"):
         requests.packages.urllib3.disable_warnings(requests.packages.urllib3.exceptions.InsecureRequestWarning)
 
-        self.api_id = os.getenv("ELASTIC_API_ID")
-        self.api_key = os.getenv("ELASTIC_API_KEY")
-        self.es_server = server or os.getenv("ELASTIC_SERVER", "https://sv-pr-elastic01:9200") # set to GSTT server by default
+        self.es_server = os.getenv("ELASTIC_SERVER", "https://sv-pr-elastic01:9200") # set to GSTT server by default
 
-        self.proxy_node = GsttProxyNode
+        if proxy:
+            self.proxy_node = GsttProxyNode
+        else:
+            self.proxy_node = None
 
-        self.es = self.create_session()
+        if conn_mode == "API":
+            self.api_id = os.getenv("ELASTIC_API_ID")
+            self.api_key = os.getenv("ELASTIC_API_KEY")
 
-    def create_session(self):
-        return Elasticsearch(
-            hosts=self.es_server,
-            api_key=(self.api_id, self.api_key),
-            node_class=self.proxy_node,
-            verify_certs=False,
-            ssl_show_warn=False
-        )
+            self.es = Elasticsearch(
+                hosts=self.es_server,
+                api_key=(self.api_id, self.api_key),
+                node_class=self.proxy_node,
+                verify_certs=False,
+                ssl_show_warn=False
+            )
+
+        if conn_mode == "HTTP":
+            self.es_user = os.getenv("ELASTIC_USER")
+            self.es_pwd = os.getenv("ELASTIC_PWD")
+
+            self.es = Elasticsearch(
+                hosts=self.es_server,
+                basic_auth=(self.es_user, self.es_pwd),
+                verify_certs=False,
+                ssl_show_warn=False
+            )
```

Comment:
And something like this

```suggestion
                self.es = Elasticsearch(
                    hosts=self.es_server,
                    api_key=(self.api_id, self.api_key),
                    node_class=self.proxy_node,
                    verify_certs=False,
                    ssl_show_warn=False
                )

            elif conn_mode == "HTTP":
                self.es_user = os.getenv("ELASTIC_USER")
                self.es_pwd = os.getenv("ELASTIC_PWD")
                
                if not all([self.es_user, self.es_pwd]):
                    raise ValueError("Check ELASTIC_USER and ELASTIC_PWD are in env variables")
                        
                self.es = Elasticsearch(
                    hosts=self.es_server,
                    basic_auth=(self.es_user, self.es_pwd),  # http_auth has been deprecated
                    verify_certs=False,
                    ssl_show_warn=False
                )
                
            else:
                raise ValueError("Argument conn_mode must be 'HTTP' or 'API'")    
```